Attribute Analysis.ipynb

```diff
@@ -0,0 +1,650 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Attribute\n",
+    "Fill in below with information about attribute you would like to analyze\n",
+    "<a id=\"attribute\"></a>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "attr_names = dataset.attribute_names\n",
+    "num_attrs = len(attr_names)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# (M1) Metric: Distance from center, size, attribute label inference\n",
+    "<a id=\"metric1\"></a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setup\n",
+    "Fill in below with number of attribute values (e.g. 2 possible values), the assigned labels names, and which attribute you want the metric to focus on (e.g. female)\n",
+    "<a id=\"metric1_setup\"></a>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "hide_toggle(for_next=True, toggle_text='Show/hide M1 code')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "info = pickle.load(open(\"results/{}/1.pkl\".format(folder_name), \"rb\"))\n",
+    "sizes = info['sizes']\n",
+    "dists = info['distances']\n",
+    "tiny_sizes = info['tiny_sizes']\n",
+    "no_faces = info['noface_sizes']\n",
+    "        \n",
+    "scenes = [None]*num_attrs\n",
+    "for attr in range(num_attrs):\n",
+    "    try:\n",
+    "        scenes[attr]=np.array(list(itertools.chain.from_iterable([chunk[2] for chunk in no_faces[attr]])) + list(itertools.chain.from_iterable([chunk[1] for chunk in tiny_sizes[attr]])))\n",
+    "    except TypeError:\n",
+    "        if len(tiny_sizes)==0:\n",
+    "            print(\"Tiny sizes is empty for {0}\".format(attr_names[attr]))\n",
+    "        else:\n",
+    "            print(\"No faces is empty for {0}\".format(attr_names[attr]))\n",
+    "    tiny_sizes[attr] = [chunk[0] for chunk in tiny_sizes[attr]]\n",
+    "    no_faces[attr] = [chunk[0] for chunk in no_faces[attr]]\n",
+    "    \n",
+    "info = pickle.load(open('util_files/places_scene_info.pkl', 'rb'))\n",
+    "idx_to_scene = info['idx_to_scene']\n",
+    "idx_to_scenegroup = info['idx_to_scenegroup']\n",
+    "sceneidx_to_scenegroupidx = info['sceneidx_to_scenegroupidx']\n",
+    "\n",
+    "xaxis = [idx_to_scenegroup[i] for i in range(len(idx_to_scenegroup))]\n",
+    "xaxis = ['\\n'.join(textwrap.wrap(chunk, width=30)) for chunk in xaxis]\n",
+    "barWidth = .4\n",
+    "fontsize = 15\n",
+    "\n",
+    "r1 = np.arange(len(idx_to_scenegroup))\n",
+    "r1 = r1 * ((barWidth * num_attrs) + .2)\n",
+    "\n",
+    "scenes = [np.bincount(scenes[i]) for i in range(num_attrs)]\n",
+    "total_image_byscene = np.sum(scenes, axis = 0)\n",
+    "total_images = np.sum(total_image_byscene)\n",
```

Comment:
you never use the variable "total_image_byscene" anywhere else, couldn't you just replace these two lines with total_images = np.sum(scenes)?

Reply:
also, previous code assumed the attributes were mutually exclusive and every image had one attribute, which is why summing the male scenes and female scenes gave you total number of scenes. we no longer want to make this assumption when attributes can hold multiple values, so perhaps you could grab the true total_images from elsewhere, rather than just adding up each attribute's number of scenes